apps/designer/app/canvas/shared/instance.ts

```diff
@@ -193,6 +193,15 @@ export const usePublishSelectedInstanceData = (treeId: Tree["id"]) => {
   const [selectedElement] = useSelectedElement();
   const [allUserProps] = useAllUserProps();
 
+  // trigger style recomputing every time
+  // new updates for this node are sent from designer
+  const [styleKey, recomputeStyles] = useReducer((d) => d + 1, 0);
+  useSubscribe("updateStyle", ({ id }) => {
```

Comment:
still kindof not guaranteed that the styles have been applied when this even arrives, because if the logic delays the application, we won't have computed the style, I think you need something that will tell you "styles have been applied"

Reply:
We already have a lot of indirection here. Let's not make it worse.
I don't see other proper way to do this without huge refactoring.
We just need to switch to data and rely less on computed styles.

Reply:
I agree in general, though we have use cases we can't get rid of, they will keep reading from the dom, so we need to have a general plan on how we organize the reads like this after updates

This updateStyle is tricky because it doesn't set style directly, it triggers tree update and react cycle, though we don't know when a specific component is going to update exactly to read from it.

I think we could have a sync event triggered by each component update from useLayutEffect specifically to allow other places like this one to read after that event has happened. 

I am ok with merging this one, but we need to start thinking how to fix the problem in a good way for all cases that need to read from the dom.